woocommerce/woocommerce
# Fix PHP warning when users have admin permissions, but no WooCommerce admin permissions and when menu items are created with no position

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




These changes fix the PHP warning shown in https://github.com/woocommerce/woocommerce/issues/38858. Previously the default page options set $args['position'] to null, but then tries to round() this variable resulting in a PHP warning.

The pull request changes the default position to 0 which can be rounded without warnings.

This situation can occur when a WooCommerce menu is created without a position such as when the woocommerce-home menu item is created for users without WooCommerce admin permissions (line 168 in Automattic\WooCommerce\Internal\Admin\Homescreen)

Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure PHP error reporting is on
2. Install WordPress and WooCommerce on your hosting platform
3. Once fully installed, log in to WordPress as an administrator and go To Admin > Users
4. Click Add New and enter a username, email and password for a new user
5. Under Role, select "Contributor"
6. Click Add New User
7. In an incognito browser window visit your WordPress website and log in as the newly created user
8. When viewing the WordPress dashboard without this PR applied you will see a PHP error saying "Deprecated: round(): Passing null to parameter # 1 ($num) of type int|float is deprecated..."
9. With this PR applied, no warning is shown



### Changelog entry



-   [ x ] Automatically create a changelog entry from the details below.

<details>

#### Significance



-   [ x ] Patch

#### Type



-   [ x ] Fix - Fixes an existing bug

#### Message 
Fixes PHP error when WooCommerce menus are created without a position and when users view the WordPress admin dashboard without having WooCommerce admin permissions.

#### Comment 

</details>
